### ISSUE TITLE
Experience Heron + MapfishPrint3 

### ISSUE THREAD

**Issue by Michael-81** [heron-mc/heron-mc#487]
Is it possible to use Mapfish Print 3 with Heron? 
Has anyone had any experience with it?

**Comment by justb4** [comment]
No, not yet, [MF Print 3](http://mapfish.github.io/mapfish-print-doc/)  (MFP3) is a total re-architecture with a different approach ([JasperReports](https://en.wikipedia.org/wiki/JasperReports)) /protocol (a.o. "templates"). See [MFP3 Presentation](https://www.slideshare.net/camptocamp/mapfish-print-3). I have done some work on MFP3 in other projects and provided a handy Docker image: https://github.com/justb4/docker-mfprint.

The current Heron Widgets (e.g. `PrintPreviewWindow`) for MFPrint2 are quite advanced with many options, also thanks to the work of GeoExt-folks and @wolwin. Will be quite some effort to reach that level. On the other hand MFP3 has a simpler approach (templates, with many predefined settings), so maybe a start could be made with a very simple MFP3 Widget.

GeoExt version 3 (Heron is based on Geoext v2) has MFP3 support: see example https://rawgit.com/geoext/geoext3/master/examples/print/basic-mapfish.html so maybe we can backport something like the [MapfishPrintProvider.js](https://github.com/geoext/geoext3/blob/master/src/data/MapfishPrintProvider.js).
